### ISSUE TITLE
Error response from daemon: Cannot start container <Contianer ID>: [8] System error: no such file or directory

### ISSUE THREAD

**Issue by drasamsetti** [boot2docker/boot2docker#1011]
I really did not understand what is this error is all about and how to resolves this..
is there anybody who faced this error and resolved this. thx

**Comment by jakirkham** [comment]
Could you give us more info like what command you ran?

**Comment by drasamsetti** [comment]
working on boot2docker env
$ docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): windows/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64

$ docker info
Containers: 3
Images: 24
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 30
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.0.7-boot2docker
Operating System: Boot2Docker 1.7.1 (TCL 6.3); master : c202798 - Wed Jul 15 00:16:02 UTC 2015
CPUs: 4
Total Memory: 1.955 GiB
Name: boot2docker
ID: JEW2:HWU5:OFSU:2UYP:VYRG:PJSP:ZMBH:C73F:RVRV:MWB2:AAJD:Y5G3
Debug mode (server): true
File Descriptors: 11
Goroutines: 17
System Time: 2015-07-29T12:11:36.452684437Z
EventsListeners: 0
Init SHA1:
Init Path: /usr/local/bin/docker
Docker Root Dir: /mnt/sda1/var/lib/docker

I am really clueless on what is going with boot2docker 
clone any docker based repo, build and run it .. ( tried mysql, mongo from tutumcloud  without any changes ) and it failed to initialise the container..

![153f026980d85539](https://cloud.githubusercontent.com/assets/6514352/8957116/bc288af0-3619-11e5-95e8-18380d7af472.png)

**Comment by jakirkham** [comment]
Are you able to run this `docker run hello-world`?

**Comment by jakirkham** [comment]
Also, are you able to share the `Dockerfile`? If so, that could be really helpful.

**Comment by jakirkham** [comment]
Can you also run `boot2docker version` and provide its results with your versions above?

**Comment by drasamsetti** [comment]
yes.. surprisingly .. i am able to run hello-world
-------------------------------------------------------------------------------------
$ docker run hello-world
Hello from Docker.
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (Assuming it was not already locally available.)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

For more examples and ideas, visit:
 http://docs.docker.com/userguide/
-----------------------------------------------------------------------------------------------
sure.. it is available at http://github.com/tutumcloud/mysql/5.6 or /5.5
http://github.com/tutumcloud/mongodb/3.0

I have not modified any bit of the Dockerfile after cloning..

**Comment by drasamsetti** [comment]
$ boot2docker version
Boot2Docker-cli version: v1.7.1
Git commit: 8fdc6f5

**Comment by jakirkham** [comment]
Could you try running `boot2docker ssh` and then try rerunning the same command to start the container?

**Comment by drasamsetti** [comment]
![4f27d91f7ab737c5](https://cloud.githubusercontent.com/assets/6514352/8957853/e52acd46-361e-11e5-9463-ea1caafdb713.png)

**Comment by jakirkham** [comment]
Could you show how you built these images?

**Comment by drasamsetti** [comment]
from the respective Docker file folder..
docker build -t drasamsetti/mysql .

**Comment by jakirkham** [comment]
This may be related to this issue ( https://github.com/docker/docker/issues/14102 ).

**Comment by drasamsetti** [comment]
after multiple uninstalls and re-installs of boot2docker .. it is still behaving in similar fashion..
i am clueless :)

**Comment by jakirkham** [comment]
Interesting, I will have to try this on a Windows machine. I have tried building and running containers based on what you have provided me on my Mac and have not been able to reproduce the error.

**Comment by jakirkham** [comment]
Actually, I won't be able to because of this ( boot2docker/boot2docker#999 ).

**Comment by jacobtomlinson** [comment]
Just my two cents, I had this error when trying to mount a volume with a typo in the path. Fixed the typo and the error went away.

**Comment by hmalphettes** [comment]
@jacobtomlinson many thanks to you for the hint.

**Comment by fischerman** [comment]
I faced the same issue on Docker Machine. If you run from Windows make sure that all shell scripts have the correct line endings (Unix). Git checks out files Windows-style on Windows. If you try to run scripts with the wrong line ending, you see something like this:
```
bash: /entrypoint.sh: /bin/bash^M: bad interpreter: No such file or directory
```
If your ENTRYPOINT points to such a script, you get the error:
```
Error response from daemon: Cannot start container 102a9cbf68a19177ed378d37f1963a454eba4d17fcef3f5e7bcae0503cc25a19: [8] System error: no such file or directory
```

**Comment by tahb** [comment]
Had the same thing, fixed it by escaping values in the `docker-compose.yml`

**Comment by jakirkham** [comment]
Yeah, these last few comments are insightful, @drasamsetti. So, the issue might be related to the fact that you are referencing some file in the docker image that does not exist. Assuming this is the cause, it either does not exist because it was not created or because of the some weird behavior in terms of how things get escaped on Windows. Something to play around with at least.
